### PR TITLE
Adding `starts_with` and `ends_with`

### PR DESCRIPTION
**Pull request by drowoseque**
## Test plan 

Unit tests provided

```
== RELEASE NOTES ==

General Changes
* Add string functions :func:`starts_with` and :func:`ends_with` 
```

### CODE REVIEW

**Comment by wanglinsong** [comment]
Please squash updates into one commit.

**Comment by drowoseque** [comment]
@wanglinsong done

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java [hunk 1]
```diff
@@ -1045,4 +1045,26 @@ public void testCharConcat()
...
+
+    @Test
+    public void testStartsWith()
+    {
+        assertFunction("starts_with('abcd', 'ab')", BOOLEAN, true);
+        assertFunction("starts_with('abcd', '')", BOOLEAN, true);
+        assertFunction("starts_with('abcd', 'ba')", BOOLEAN, false);
+        assertFunction("starts_with('', 'ba')", BOOLEAN, false);
+        assertFunction("starts_with(NULL, 'ba')", BOOLEAN, null);
+        assertFunction("starts_with('abcd', NULL)", BOOLEAN, null);
```
Comment (feilong-liu) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java:1057]: LGTM, can you add a few more test cases like
(NULL, NULL)
('',  NULL)
(NULL, '')

Similar for the endswith tests.

**Review by feilong-liu — APPROVED** [review]

**Review by pranjalssh — APPROVED** [review]
LGTM

**Pull request merged**